src/wallet/test/silentpayment_tests.cpp

```diff
@@ -0,0 +1,172 @@
+#include <addresstype.h>
+#include <wallet/silentpayments.h>
+#include <test/data/bip352_send_and_receive_vectors.json.h>
+
+#include <test/util/setup_common.h>
+#include <hash.h>
+
+#include <boost/test/unit_test.hpp>
+#include <test/util/json.h>
+#include <vector>
+#include <util/bip32.h>
+#include <wallet/wallet.h>
+
+namespace wallet {
+BOOST_FIXTURE_TEST_SUITE(silentpayment_tests, BasicTestingSetup)
+
+CKey ParseHexToCKey(std::string hex) {
+    CKey output;
+    std::vector<unsigned char> hex_data = ParseHex(hex);
+    output.Set(hex_data.begin(), hex_data.end(), true);
+    return output;
+};
+
+CKey GetKeyFromBIP32Path(std::vector<std::byte> seed, std::vector<uint32_t> path)
+{
+    CExtKey key;
+    key.SetSeed(seed);
+    for (auto index : path) {
+        BOOST_CHECK(key.Derive(key, index));
+    }
+    return key.key;
+}
+
+BOOST_AUTO_TEST_CASE(bip352_send_and_receive_test_vectors)
+{
+    UniValue tests;
+    tests.read(json_tests::bip352_send_and_receive_vectors);
+
+    for (const auto& vec : tests.getValues()) {
+        // run sending tests
+        BOOST_TEST_MESSAGE(vec["comment"].get_str());
+        for (const auto& sender : vec["sending"].getValues()) {
+            const UniValue& given = sender["given"];
+            const UniValue& expected = sender["expected"];
+
+            std::vector<COutPoint> outpoints;
+            for (const auto& outpoint : given["outpoints"].getValues()) {
+                outpoints.emplace_back(uint256S(outpoint[0].get_str()), outpoint[1].getInt<uint32_t>());
+            }
+
+            std::vector<std::pair<CKey, bool>> sender_secret_keys;
+            for (const auto& key : given["input_priv_keys"].getValues()) {
```

Comment:
With current implementation, we can't have test vectors for sending that include non-eligible sp inputs. This is because we just sum all the private keys without checking if the input is eligible. I propose to extract eligibility checking as a function from `GetSilentPaymentsTweakDataFromTxInputs` so we can use it here in the unit test.